BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2776,6 +2779,54 @@ public override Vector3 GetForwardTransform()
                 return MissileReferenceTransform.forward;
         }
 
+        public override float GetKinematicTime()
+        {
+            // Get time at which the missile is traveling at the GetKinematicSpeed() speed
+            if (!launched) return -1f;
+
+            float missileKinematicTime = boostTime + cruiseTime + cruiseDelay + dropTime - TimeIndex;
+            float speed = currentThrust > 0 ? optimumAirspeed : (float)vessel.srfSpeed;
+            float minSpeed = GetKinematicSpeed();
+            if (speed > minSpeed)
+            {
+                float airDensity = (float)vessel.atmDensity;
+                float dragTerm;
+                float t;
+                if (useSimpleDrag)
+                {
+                    dragTerm = (deployed ? deployedDrag : simpleDrag) * (0.008f * part.mass) * 0.5f * airDensity;
+                    t = part.mass / (minSpeed * dragTerm) - part.mass / (speed * dragTerm);
+                }
+                else
+                {
+                    float AoA = smoothedAoA.Value;
+                    FloatCurve dragCurve = MissileGuidance.DefaultDragCurve;
+                    float dragCd = dragCurve.Evaluate(AoA);
+                    float dragMultiplier = BDArmorySettings.GLOBAL_DRAG_MULTIPLIER;
+                    dragTerm = 0.5f * airDensity * dragArea * dragMultiplier * dragCd;
+                    float dragTermMinSpeed = 0.5f * airDensity * dragArea * dragMultiplier * dragCurve.Evaluate(Mathf.Min(29f, maxAoA)); // Max AoA or 29 deg (at kink in drag curve)
+                    t = part.mass / (minSpeed * dragTermMinSpeed) - part.mass / (speed * dragTerm);
+                }
+                missileKinematicTime += t; // Add time for missile to slow down to min speed
+            }
+
+            return missileKinematicTime;
+        }
+
+        public override float GetKinematicSpeed()
+        {
+            // Get speed at which the missile is only capable of pulling a 2G turn at maxAoA
+            float Gs = 2f;
+
+            FloatCurve liftCurve = MissileGuidance.DefaultLiftCurve;
```

Comment:
`DefaultLiftCurve` is `null` before `DoAeroForces` gets called.
Is there any reason for `DefaultLiftCurve` and `DefaultDragCurve` to be initially `null` instead of
```csharp
public static FloatCurve DefaultLiftCurve = new([
    new (0, 0),
    new (8,0.35f),
    new (30, 1.5f),
    new (65, 0.6f),
    new (90, 0.7f)
]);
public static FloatCurve DefaultDragCurve = new([
    new (0, 0.00215f, 0.00014f, 0.00014f),
    new (5, .00285f, 0.0002775f, 0.0002775f),
    new (15, .007f, 0.0003146428f, 0.0003146428f),
    new (29, .01f, 0.0002142857f, 0.01115385f),
    new (55, .3f, 0.008434067f, 0.008434067f),
    new (90, .5f, 0.005714285f, 0.005714285f)
]);
```